pallets/collator-selection/src/lib.rs

```diff
@@ -355,9 +376,15 @@ pub mod pallet {
 			// `reward` is half of treasury account, this should never fail.
 			let _success = T::Currency::transfer(&treasury, &author, reward, KeepAlive);
 			debug_assert!(_success.is_ok());
-
+			<Candidates<T>>::mutate(|candidates| {
+				if candidates.len() > 0 {
+				let index: usize = candidates.iter().position(|candidate| candidate.who == author).unwrap_or_default();
```

Comment:
```suggestion
				let index: usize = candidates.iter().position(|candidate| candidate.who == author).unwrap_or_default();
```

If the candidate is not found, why should we remove the first one? We should skip if it does not exists.

Note that the author might actually no longer be a candidate in our code, because of session buffering. So it is totally legit that an author is no longer candidate in our pallet and should be handled better.